api/graphql/schema.graphql

```diff
@@ -1131,6 +1135,10 @@ input CreateTeamInput {
 
 input UpdateTeamInput {
   name: String
+    @rules(
+      apply: ["unique:teams,name"]
+      messages: [{ rule: "unique", message: "TeamNameInUse" }]
+    )
```

Comment:
This blocks updating a team even if the name stays the same. 
Validating updates I believe requires the validation class with ignores like so. 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/830c1050-5948-47e3-a246-c4b19caba958)

Reply:
Ah, doesn't quite work yet

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/5a199750-a21c-411d-996d-3960a94589c4)

Will need to go on the mutation itself as you need the validator to be able to see the input ID. For this mutation the id is not part of the input object. 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/a28d3930-0ed3-411e-a758-361276088407)

While the other ones the id is part of the input object so you can stick the validator on that. This is the pattern I think we want going forward but refactoring existing mutations isn't something to worry about yet. 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/3add5abe-5dbc-459f-8877-393fb891b296)

Reply:
The same applies to the updateDepartment mutation

Reply:
Oh, yeah. That makes sense. Moved in [move validators to top level mutations](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8471/commits/e3a876d13236fe2e475b1bdd1f62fc914344ffa1)